ks_patches/set_assigned_user.js

```diff
@@ -0,0 +1,19 @@
+export function setAssignedUser(teiId, assignedUser, programId, programStage, DHIS2EventFactory, TEIService, DateUtils) {
+    TEIService.getTeiWithAllAvailableFields(teiId, null, []).then((tei) => {
+        if (tei && tei.enrollments) {
+            var enrollment = tei.enrollments.find(enrollment => enrollment.program === programId && enrollment.status === 'ACTIVE');
+            if (enrollment && enrollment.events) {
+                var event = enrollment.events.find(event => event.programStage === programStage && event.status !== 'COMPLETED');
```

Comment:
Can the event also be scheduled? What is the desired behavior if it is?

Reply:
The event can be anything but 'COMPLETED',  (mark the !==). In those three cases the user should be assigned.
It is, unfortunately only implicitly, guaranteed from the logic allowing the user to set assigned user that there will be one and only one event with status not COMPLETED. So that one not COMPLETED event will get assigned to the selected user